### DISCUSSION TITLE
issue: "onBlur" validation state not in sync with data via watch()

### DISCUSSION THREAD

**Discussion by maxigs** [react-hook-form/react-hook-form#9610] [Q&A]
### Version Number

7.41.0

### Codesandbox/Expo snack

https://codesandbox.io/s/young-surf-38qvs0?file=/src/App.tsx

### Steps to reproduce

1. Open console log
2. Enter `1` in the field : it logs `{firstName: '1'}` as valid data, correctly
3. Update field to `a` : it log `{firstName: 'a'}` as valid data, **incorrectly**
4. Exit the field input, causing `onBlur` event
5. Update field to `ab` :  no log output appears, correctly

When adding `mode: 'onChange'` to form declaration it works correctly.

It seems, the `watch()` triggers and returns the updated form data before the validation has taken place, which causes the `formState` to be out of sync with the data of the form.

### Expected behaviour

The form should not return to be valid, if there is a known data change that has not been validated yet.
Possibly `watch()` should trigger with the same `mode` as the validations are excuted by default.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`watch` is only subscribed to change events, that's what it is made for, and the watch result is recommended not to be used at the `useEffect` dep (according to the doc), perhaps you want to add your logic attached to `onBlur` event instead.

**Comment by maxigs** [comment]
This was the example i followed:
https://github.com/react-hook-form/react-hook-form/discussions/7063#discussioncomment-1657441 

Not sure if there is a better way implement the functionality since that post. I did convert it to `useWatch` instead of `watch()` already and it works quite well – once  the `onChange` mode was set.

Not sure if it's really a bug in code, or issue due to uncommon use, but it seemed noteworty.

**Comment by bluebill1049** [comment]
Thanks, @maxigs for pointing that out, I belive in the example, it's an `onChange` event in the example instead of onBlur, watch API currently only subscribe to on change. I would still refer to event handler before go after `useEffect`
